addons/pandora/ui/editor/pandora_editor.gd

```diff
@@ -171,3 +179,49 @@ func _data_load_failure() -> void:
 	data_content.visible = false
 	error_content.visible = true
 	_load_error = true
+
+func _import_file(path: String) -> void:
+	Pandora.calculate_import_data(path)
+
+func _on_import_calculated(import_info: Dictionary) -> void:
```

Comment:
The `pandora_editor` is supposed to be rather independent of specific functionality and should only "orchestrate" between various bits.

Move the "import dialog" into its own scene with its own dedicated nodes and use it here, e.g. on top:
```gdscript
@onready var import_dialog = %PandoraImportDialog
```
and then in `_ready():`
```gdscript
import_button.pressed.connect(import_dialogue.popup_centered)
```
All the other logic can then sit in the import dialog.

Having nodes visually in its own scene is much better to tweak things visually rather than it being created in GDScript (I find)